tensorboard/components/vz_line_chart2/line-chart-exporter.ts

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace vz_line_chart2 {
+
+enum NodeName {
+  GROUP = 'G',
+  DIV = 'DIV',
+  SVG = 'SVG',
+  TEXT = 'TEXT',
+}
+
+export class PlottableExporter {
+  private root: Element;
+  private uniqueId: number = 0;
+
+  constructor(rootEl: Element) {
+    this.root = rootEl;
+  }
+
+  public asString(): string {
```

Comment:
Nit, but `asString()` seems like not quite the right name because to me it communicates turning the exporter itself into a string, which isn't what happens here.  Something like `exportAsString()` or `exportToString()` or `createString()` would seem clearer.